### ISSUE TITLE
dopeflamingo - When ```block.timestamp == allocationEndTime``` this can bypassed both ```onlyActiveAllocation()``` and ```onlyAfterAllocation()``` modifiers.

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-09-Gitcoin-judging#839]
dopeflamingo

medium

# When ```block.timestamp == allocationEndTime``` this can bypassed both ```onlyActiveAllocation()``` and ```onlyAfterAllocation()``` modifiers.
On the condition that ```block.timestamp == allocationEndTime``` both modifiers can be bypassed as neither of them revert on this condition. This can lead to unwanted behaviour as it is unclear whether the protocol wants this specific time to be part of the allocation process or not.

## Vulnerability Detail
The distribute based functions should only be called after all the allocation has taken place and the allocate functions should only be called during active allocation times, however this specific condition at  ```block.timestamp == allocationEndTime``` has left it so that both functions can be called at the same time. 

## Impact
This could lead to a distribution function being called before an allocation has taken place which seems to be unwanted behaviour for this protocol.

## Code Snippet

https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/donation-voting-merkle-base/DonationVotingMerkleDistributionBaseStrategy.sol#L465-L469

https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/donation-voting-merkle-base/DonationVotingMerkleDistributionBaseStrategy.sol#L473-L476

These modifiers are also used in QVBaseStrategy.sol.

## Tool used

Manual Review

## Recommendation

I would recommend that the protocol make it so that the ```_checkOnlyAfterAllocation()``` function reverts at the time ```block.timestamp == allocationEndTime```. 

Change the if statement from ```if (block.timestamp < allocationEndTime)``` to:

```if (block.timestamp <= allocationEndTime)``` so that it is strictly only after the allocation period has ended.

**Comment by jkoppel** [comment]
Escalate.

Of the 5 issues marked as duplicate of this (including this present report), only 2 identify an actual issue.

This one, #839, does not identify a negative consequence of the 1-second overlap between allocation and distribution time. See its impact statement, "This could lead to a distribution function being called before an allocation has taken place which seems to be unwanted behaviour for this protocol."

#184  does not either. See its impact statement, "The status of the allocation becomes ambiguous, compromising the integrity of the protocol's functionality."

Both should be marked invalid. According to Sherlock rules, "If the submission is vague or does not identify the attack path with higher severity clearly it will be considered low." https://docs.sherlock.xyz/audits/judging/judging

#924  has nothing to do with this report. I have already escalated it separately.

In contrast, both #835  and #261 show actual harm that results from this issue.

According to Sherlock rules, if there are submissions that identify a valid attack path from an issue, but also "there is a submission D which identifies the core issue but does not clearly describe the impact or an attack path. Then D is considered low."  https://docs.sherlock.xyz/audits/judging/judging

Thus, this issue should be awarded to #835  and #261 as the only two valid reports of this issue. #839, #184, and #924 should marked low or invalid.

**Comment by sherlock-admin2** [comment]
> Escalate.
> 
> Of the 5 issues marked as duplicate of this (including this present report), only 2 identify an actual issue.
> 
> This one, #839, does not identify a negative consequence of the 1-second overlap between allocation and distribution time. See its impact statement, "This could lead to a distribution function being called before an allocation has taken place which seems to be unwanted behaviour for this protocol."
> 
> #184  does not either. See its impact statement, "The status of the allocation becomes ambiguous, compromising the integrity of the protocol's functionality."
> 
> Both should be marked invalid. According to Sherlock rules, "If the submission is vague or does not identify the attack path with higher severity clearly it will be considered low." https://docs.sherlock.xyz/audits/judging/judging
> 
> #924  has nothing to do with this report. I have already escalated it separately.
> 
> In contrast, both #835  and #261 show actual harm that results from this issue.
> 
> According to Sherlock rules, if there are submissions that identify a valid attack path from an issue, but also "there is a submission D which identifies the core issue but does not clearly describe the impact or an attack path. Then D is considered low."  https://docs.sherlock.xyz/audits/judging/judging
> 
> Thus, this issue should be awarded to #835  and #261 as the only two valid reports of this issue. #839, #184, and #924 should marked low or invalid.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by deadrosesxyz** [comment]
Escalate. 

Issue #546 is a valid duplicate of issue #839. It has been duped to a unrelated issue and incorrectly excluded.  Also it provides PoC code that displays potential impact as per jkoppel escalation comment. Please validate the issue and consider High impact since funds can end up stuck permanently & core protocol assumptions are broken around allocate() & distribute() order of invocation.

**Comment by sherlock-admin2** [comment]
> Escalate. 
> 
> Issue #546 is a valid duplicate of issue #839. It has been duped to a unrelated issue and incorrectly excluded.  Also it provides PoC code that displays potential impact as per jkoppel escalation comment. Please validate the issue and consider High impact since funds can end up stuck permanently & core protocol assumptions are broken around allocate() & distribute() order of invocation. 
> 

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by neeksec** [comment]
Agree with Escalation that

#835, #261 and #546 are valid submissions since they all identified the bug and provided valid impact description.

#839, #184 should be excluded because of no valid impact description. #924 is not a dup and is invalid.

Suggest to keep Medium because the locking has precondition that allocate and distribute happen in one block which is a rare case.

**Comment by jacksanford1** [comment]
https://github.com/allo-protocol/allo-v2/pull/337

**Comment by Evert0x** [comment]
Planning to accept escalations, make #839, #184 and #924 invalid. Make #835 #261 and #546 a medium family.

**Comment by Evert0x** [comment]
Result:
Invalid 
Unique

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [jkoppel](https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues/839/#issuecomment-1762821673): accepted
- [deadrosesxyz](https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues/839/#issuecomment-1762924342): accepted
